helix-term/src/commands.rs

```diff
@@ -5982,6 +5987,12 @@ fn jump_to_label(cx: &mut Context, labels: Vec<Range>, behaviour: Movement) {
             doc_mut!(cx.editor, &doc).remove_jump_labels(view);
             return;
         }
+        if let Some(ch) = event.char() {
```

Comment:
We can figure out the overlays with just `outer` though instead of keeping around a hashmap. The new overlays at this point should be created from `labels().iter().skip(outer).take(alphabet.len())`.

In order to make the borrowing work you will need to introduce a helper function for `alphabet_char` that borrows the alphabet instead of the current `alphabet_char` that borrows the alphabet from the closure's environment:

```rust
fn alphabet_char(alphabet: &[char], index: usize) -> Tendril {
    let mut res = Tendril::new();
    res.push(alphabet[index]);
    res
}
```

With this you can remove the hashmap and the changes to `Overlay`